tests/mmm/test_delayed_saturated_mmm.py

```diff
@@ -579,6 +610,63 @@ def test_model_config(
                 model.model["beta_channel"].owner.op, pm.HalfNormal
             )  # beta_channel
 
+    @pytest.mark.parametrize(
+        "model_name", ["mmm_fitted", "mmm_fitted_with_fourier_features"]
+    )
+    @pytest.mark.parametrize(
+        "new_dates",
+        [
+            # 2021-12-31 is the last date in the toy data
+            # Old and New dates
+            pd.date_range("2021-11-01", "2022-03-01", freq="W-MON"),
+            # Only old dates
+            pd.date_range("2019-06-01", "2021-12-31", freq="W-MON"),
+            # Only new dates
+            pd.date_range("2022-01-01", "2022-03-01", freq="W-MON"),
+            # Less than the adstock_max_lag (4) of the model
+            pd.date_range("2022-01-01", freq="W-MON", periods=1),
+        ],
+    )
+    @pytest.mark.parametrize("original_scale", [True, False])
+    def test_new_data_predictions(
+        self,
+        generate_data,
+        model_name: str,
+        new_dates: pd.DatetimeIndex,
+        original_scale: bool,
+        request,
+    ):
+        mmm = request.getfixturevalue(model_name)
+        X_pred = generate_data(new_dates)
+
+        pp_without = mmm.predict_posterior(
```

Comment:
@juanitorduz 
I can still break these down a bit more
I'm trying to get a handful of the edge cases via the parametrize tests

Are you thinking one for each predict method?

Reply:
Yes, ideally test should be atomic (test one thing) and have all assert statements at the end. Not a strong opinion, more.of a suggestion :)

Reply:
Fair enough. I'm pulling out the date ranges and separating the tests